### ISSUE TITLE
Jumpy listview

### ISSUE THREAD

**Issue by JesseBIT** [kmcgill88/admob_flutter#181]
I have a listview.builder working of a firestore querysnapshot stream that builds a Card. Works fine and really smooth, but I wanted to add a Admob after every 3 cards so this was the perfect tool. I've done it by using a column in listbuilder, with the card created from the firestore data and then the admob below it if it's index is every 3rd. The list is so jumpy/jittery everytime scroll past an ad. Disable the Ads and it's perfectly smooth again. Anyone else had this issue, seems to be as it's disposing the ad's when they go out of view. Such a shame as a show stopper for me being able to use this.

**Comment by JanNovakFRI** [comment]
Hello @JesseBIT, I have the same problem as you. Now I am playing with the source code and example code to find solution. I managed to stop ads from being destroyed by adding `AutomaticKeepAliveClientMixin` to `AdmobBanner` class in source lib and `addAutomaticKeepAlives: true`, `cacheExtent` to listview builder. Now once an ad is loaded scrolling up is smooth :), but scrolling down and loading new ads is still jumpy. Do you have any idea what would be the cause?

**Comment by felixgabler** [comment]
I have the same issue.

**Comment by kmcgill88** [comment]
Per [Admob policy](https://support.google.com/admob/answer/6275345?hl=en&ref_topic=2936214), adding banners to scroll areas is a policy violation.

`banner ads should not move as a user scrolls, as users may try to click on the menu but end up clicking on the ad accidentally instead. This specific implementation is against policy and we reserve the right to disable ad serving to your app.`

Discussed here: https://github.com/kmcgill88/admob_flutter/issues/194

I am actively working on refactoring the demo app to follow encouraging implementations.

**Comment by JesseBIT** [comment]
what a shame. So many apps use that method of advertising, fb, twitter, reddit etc etc. Users are more than used to seeing Ads in scrollable content. I think it looks so much more proffesional than a static ad up top or bottom. :-(

**Comment by kmcgill88** [comment]
@JesseBIT I agree with you. I was honestly shocked to find this in the docs. If you are willing to risk the violation, you can go ahead and keep your scroll ads. Due to this policy, I won't be investigating the jumpy list. 😭 

If you can find any other official docs contradicting this AdMob policy please share.

**Comment by JesseBIT** [comment]
Reluctantly I may give this a go: https://pub.dev/packages/facebook_audience_network

Thanks for everything you done on this one @kmcgill88 . I raised the question on the official repo too.

**Comment by kmcgill88** [comment]
@JesseBIT can you please share a link to your question on their official repo? I'd like to follow that.

**Comment by JesseBIT** [comment]
Probably worded more as a rant 😬

https://github.com/FirebaseExtended/flutterfire/issues/3105

**Comment by jagomf** [comment]
> @JesseBIT I agree with you. I was honestly shocked to find this in the docs. If you are willing to risk the violation, you can go ahead and keep your scroll ads. Due to this policy, I won't be investigating the jumpy list. 😭
> 
> If you can find any other official docs contradicting this AdMob policy please share.

Just replied to #194.
IMHO the interpretation of «banner ads should not move as a user scrolls» does *not* include banners which are actually fixed to a position in a list, and thus not actually moving other than all the content.

Google themselves actually do sell that kind of banners in platforms of theirs such as Blogspot blogs.

Please @kmcgill88 do not disable the ability to place banners in lists. I find actually `addAutomaticKeepAlives` way more relevant regarding this.

**Comment by MichaelMalony** [comment]
Looks like inline ads are coming to Google's admob plugin so I don't think it's a violation: https://medium.com/flutter/new-ads-beta-inline-banner-and-native-support-for-the-flutter-mobile-ads-plugin-e48a7e9a0e64

**Comment by JesseBIT** [comment]
Thanks Michael, this is great news. I've registered for the Beta but ironically this is one of 2 issues thats prevented me publishing the app, so not sure I'll get selected :-(.
